SalesforceCommerceCloud/pwa-kit
# @W-15490276@ Improved auth error logging for PWA Kit v2

This is the v2 version of #2028 and is related to #1829 

There are a few ways in which a call to SLAS to consume a refresh token can fail. For example, the client id the refresh token is associated with might be different from the client id included with the request.

Before this change, when the refresh token call fails, we do not clear the bad refresh token from local / cookie storage. As a result, when the user tries to log in again, the system will re-try with the same bad refresh token until the refresh token expires after 90 days or is manually cleared.

This change clears the refresh token when the refresh token fails.

This change also updates logging so that calls to the SLAS /authorize, /authenticate, and /token endpoints are logging error messages when they fail.

For v2, I updated SLAS calls to include the channel_id property

Testing the changes:
Refresh login (note: the refresh login request happens client side only):
1. Start the app with these changes locally (this will start the app against zzrf-001)
2. Delete the access token in your local store
3. On a different browser window, open up https://test.phased-launch-testing.com/ and grab the refresh token
4. Copy the refresh token from test.phased-launch-testing into the cc-nx-g cookie on the tab that is open on zzrf-001 (on v2, the refresh token is on local storage by default)
5. Refresh the tab on zzrf-001
6. See in the browser console that there is a log entry for `commerce-sdk-react ERROR invalid organizationId`
7. Verify that your refresh token was replaced with a new one that is usable for zzrf-001

New guest login:
1. In default.js, set an invalid client ID
2. Start the app (make sure you don't have an existing refresh token or access token)
3. In server logs, see that there is a log entry for 400 Bad Request with a descriptive SLAS error message (ie. client_id must be 36 chars)
4. In browser console, see the same message.
